components/compositing/compositor.rs

```diff
@@ -1213,65 +765,11 @@ impl<Window: WindowMethods> IOCompositor<Window> {
-                              epoch: Epoch,
-                              frame_tree_id: FrameTreeId) {
-        // If the frame tree id has changed since this paint request was sent,
-        // reject the buffers and send them back to the paint thread. If this isn't handled
-        // correctly, the content_age in the tile grid can get out of sync when iframes are
-        // loaded and the frame tree changes. This can result in the compositor thinking it
-        // has already drawn the most recently painted buffer, and missing a frame.
-        if frame_tree_id == self.frame_tree_id {
-            if let Some(layer) = self.find_layer_with_pipeline_and_layer_id(pipeline_id,
-                                                                            layer_id) {
-                let requested_epoch = layer.extra_data.borrow().requested_epoch;
-                if requested_epoch == epoch {
-                    self.assign_painted_buffers_to_layer(layer, new_layer_buffer_set, epoch);
-                    return
-                } else {
-                    debug!("assign_painted_buffers epoch mismatch {:?} {:?} req={:?} actual={:?}",
-                           pipeline_id,
-                           layer_id,
-                           requested_epoch,
-                           epoch);
-                }
-            }
-        }
-
-        self.cache_unused_buffers(new_layer_buffer_set.buffers);
-    }
-
-    fn assign_painted_buffers_to_layer(&mut self,
-                                       layer: Rc<Layer<CompositorData>>,
-                                       new_layer_buffer_set: Box<LayerBufferSet>,
-                                       epoch: Epoch) {
-        debug!("compositor received new frame at size {:?}x{:?}",
-               self.window_size.width,
-               self.window_size.height);
-
-        // From now on, if we destroy the buffers, they will leak.
-        let mut new_layer_buffer_set = new_layer_buffer_set;
-        new_layer_buffer_set.mark_will_leak();
-
-        // FIXME(pcwalton): This is going to cause problems with inconsistent frames since
-        // we only composite one layer at a time.
-        layer.add_buffers(self, new_layer_buffer_set, epoch);
-        self.composite_if_necessary_if_not_using_webrender(CompositingReason::NewPaintedBuffers);
-    }
-
     fn scroll_fragment_to_point(&mut self,
-                                pipeline_id: PipelineId,
-                                layer_id: LayerId,
-                                point: Point2D<f32>) {
-        if self.move_layer(pipeline_id, layer_id, TypedPoint2D::from_untyped(&point)) {
-            self.perform_updates_after_scroll();
-            self.send_viewport_rects_for_all_layers()
-        } else {
-            self.fragment_point = Some(point)
-        }
+                                _pipeline_id: PipelineId,
+                                _layer_id: LayerId,
+                                _point: Point2D<f32>) {
```

Comment:
So if I understand this code correctly, this is only called on resize to keep the scroll where the fragment of the URL points to. The code in `script_thread.rs` has a `FIXME(pcwalton)` saying it's very bogus. I think it's a regression we could take, what do you think?